tests/e2e/unified_test.go

```diff
@@ -325,6 +326,11 @@ func testFixturesInSeries(ctx context.Context, t *testing.T, testPause bool, can
 				if testPause {
 					opt.SkipWaitForReady = true // Paused resources don't send out an event yet.
 				}
+				if os.Getenv("GOLDEN_REQUEST_CHECKS") != "" {
+					// If we're doing golden request checks, delete synchronously so that it is reproducible.
+					// Note that this does introduce a dependency that objects are ordered correctly for deletion.
+					opt.CreateInOrder = true
```

Comment:
Aha! The `pause` tests are expecting objects to not have conditions which, in turn, will mean they are not marked ready in our testing.